File: Codes/2239-find-closest-number-to-zero.java
/*
 * @Author: LetMeFly
 * @Date: 2025-01-20 13:33:12
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-01-20 13:33:16
 */
class Solution {
    public int findClosestNumber(int[] nums) {
        int ans = nums[0];
        for (int t : nums) {
            if (Math.abs(t) < Math.abs(ans) || Math.abs(t) == Math.abs(ans) && ans < 0) {
                ans = t;
            }
        }
        return ans;
    }
}

File: Codes/2239-find-closest-number-to-zero.go
/*
 * @Author: LetMeFly
 * @Date: 2025-01-20 13:34:23
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-01-20 16:00:01
 */
package main

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}

func findClosestNumber(nums []int) int {
    ans := nums[0]
    for _, t := range nums {
        if abs(t) < abs(ans) || abs(t) == abs(ans) && ans < 0 {
            ans = t
        }
    }
    return ans
}